MoltenVK/MoltenVK/Utility/MVKVector.h

```diff
@@ -0,0 +1,496 @@
+/*
+ * MVKVectorAllocator.h
+ *
+ * Copyright (c) 2012-2018 Dr. Torsten Hans (hans@ipacs.de)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+//
+// a simple std::vector like container with a configurable extra stack space
+// this class supports just the necessary members to be compatible with MoltenVK
+// if C++17 is used, code can be simplified further
+// by default MVKVector used 8 elements from the stack before getting memory from heap
+//
+#include "MVKVectorAllocator.h"
+#include <type_traits>
+#include <initializer_list>
+#include <utility>
+
+template<class Type, class Allocator = mvk_vector_allocator_with_stack<Type, 8>> class MVKVector
+{
+  Allocator alc;
+
+public:
+  class iterator
+  {
+    const MVKVector *vector;
+    size_t           index;
+
+  public:
+    iterator() = delete;
+    iterator( const size_t _index, const MVKVector &_vector ) : vector{ &_vector }, index{ _index } { }
+
+    iterator &operator=( const iterator &it )
+    {
+      vector = it.vector;
+      index = it.index;
+      return *this;
+    }
+
+    Type *operator->() const

```

Comment:
This iterator class is missing an `operator*`. I think that's why you were unable to use `std::copy_n()` or the `assign(begin, end)` idiom.